### ISSUE TITLE
wxStyledTextCtrl isn't accessible with MacOS voiceover screenreader

### ISSUE THREAD

**Issue by Daniil-Gusev** [wxWidgets/wxWidgets#26652]
See #18683.
The issue seems to be present on MacOS as well: wxStyledTextCtrl isn't detected by voiceover.
I'm using wxWidgets v3.3.2, a port of wx cocoa.
Any ideas?

**Comment by vadz** [comment]
Unfortunately I don't know much about accessibility under macOS so we would need help with fixing this.

**Comment by MaartenBent** [comment]
I suppose the `NSAccessibility` functions will need to be implemented somewhere, but I have no idea where (`PlatWXcocoa.mm`?) and how.
Similar to what Scite does (from line 988): https://sourceforge.net/p/scintilla/code/ci/default/tree/cocoa/ScintillaView.mm

**Comment by torstenvl** [comment]
wxAccessible is unfortunately only implemented on Windows for now. See: https://forums.wxwidgets.org/viewtopic.php?t=4351

I am interested in broadening wxWidgets Accessibility, and I've been looking into it as part of my work on wxRichTextCtrl (for which the existing wxAccessible implementation based on the MSAA/IAccessible API is insufficient even on Windows). But it's a lot of work, with a lot of legacy API and ABI concerns, and I have a very busy full-time job. 

I also don't really know enough about macOS ABI to know off the top of my head if the concerns are the same. It's POSSIBLE that the message-passing nature of ObjC means this be handled without breaking ABI the way messing with C++ object vtables can. I would need to research and think about that a lot before giving an intelligent answer.

**Comment by Daniil-Gusev** [comment]
In any case, thank you for any work you do on this. I'm unlikely to be able to help you with the code at this stage, but if needed, please feel free to mention me, and I'll be happy to test patches.

**Comment by torstenvl** [comment]
I got a minimal proof of concept for macOS using wxAccessible working last weekend. The `NSAccessibility` *implementation* shouldn't break ABI, but wiring it up does. 

<img width="2304" height="1296" alt="Image" src="https://github.com/user-attachments/assets/4664c030-454e-498d-8cb2-27edd137174c" />

(Note contents of kTextAcc set as the value in the wxAccessible-derived HelloAccessible class appearing in the VoiceOver transcript box in the lower right.)

Currently, there is an ABI break between accessible builds and non-accessible builds, because the very existence of the `wxAccessible *` on `wxWindowBase` is gated behind `wxUSE_ACCESSIBILITY`. As a result, `sizeof(wxWindow)` changes based on the compile-time options. 

@vadz I recommend that for 3.4, the `wxAccessible *` and related `virtual`s be ungated, and be part of `wxWindowBase` regardless of compile-time options. This will maintain ABI of `wxWindow`; instead, we can use `wxUSE_ACCESSIBLE` to gate the implementation methods, reducing them to stubs when accessibility is turned off and compiling the accessibility implementation in when it's turned on. This will in turn allow us to make accessibility improvements on Mac and Windows (and maybe GTK?) throughout the 3.4.x lifecycle.
